### PR TITLE
Added automatic draft and restore of rich text

### PR DESCRIPTION
**Pull request by mjvkeulen**
Rich text content is now stored in local storage by default as a draft and remains there for 1 week. If a draft can be found for a rich text area, it is automatically restored.

Note that it is important the rich text editor receives a very specific id as it serves as the local storage key for draft storage.

### CODE REVIEW

**Comment by engelhamer** [comment]
This is nice but maybe include the other fields of e.g. an Entry as well in this PR? Now it might be weird for users that their rich text content does appear after reloading the page, but not other fields.

File: src/vue/src/components/entry/EntryFields.vue [hunk 1]
```diff
@@ -65,7 +65,7 @@
                 <!-- Newly added fields in template editor have id <0. -->
                 <text-editor
                     v-else-if="field.type == 'rt'"
-                    :id="`rich-text-editor-field-${field.id > 0 ? 'id-' + field.id : 'loc-' + field.location}`"
+                    :id="textEditorID(field)"
                     :key="`rich-text-editor-field-${field.id > 0 ? field.id : field.location}`"
```
Comment (Larspolo) [inline src/vue/src/components/entry/EntryFields.vue:69]: Should key and id be the same? When is which used?

**Review by Larspolo — CHANGES_REQUESTED** [review]
1. go to format editor
2. create new template
3. insert text in first field
4. wait 10 seconds
5. save assignment
6. navigate away
7. repeat 1 and 2
8. -> content is loaded in again, think this should be erased as it is a different new entry

Same goes for when creating a new assignment, or when placing a new comment (most common probably, see screenshot), or when creating a new presetnode in the format editor

image

File: src/vue/src/components/entry/EntryFields.vue (hunk 1 above)
Comment (mjvkeulen) [reply src/vue/src/components/entry/EntryFields.vue:69]: key needs to be unique for that specific dom view, id needs to map the specific RT to storage.

Comment (Larspolo) [reply src/vue/src/components/entry/EntryFields.vue:69]: So why not keep it the same?

Comment (mjvkeulen) [reply src/vue/src/components/entry/EntryFields.vue:69]: Can do but why does that have your preference? (Either way is fine by me)

Comment (Larspolo) [reply src/vue/src/components/entry/EntryFields.vue:69]: For consistency mostly

**Review by engelhamer — COMMENTED** [review]
In my opinion this is not very usable now, since e.g. comments re-appear illogically after a refresh (like Lars pointed out above). See my previous comment for it to be useful.

**Comment by mjvkeulen** [comment]
Moving to another solution #688

**Pull request closed without merging**